src/ucs/stats/stats.h

```diff
@@ -19,10 +19,61 @@ BEGIN_C_DECLS
 
 /** @file stats.h */
 
+/* The maximum size of a class name */
+#define MAX_CLS_NAME_SIZE 50
+
+/* The maximum number of UCX aggregate-sum counters */
+#define MAX_NUM_OF_STATS_AGGREGATE_SUM_COUNTERS 32
+
+/* The maximum number of statistics classes */
+#define MAX_NUM_OF_STATS_CLASSES 32
+
+typedef struct tag_ucs_stats_counter {
+    /* Counter class name */
+    const char *class_name;
+
+    /* Counter name */
+    const char *name;
+
+    /* Counter value */
+    uint64_t value;
+
+} ucs_stats_counter_s;
```

Comment:
Note that in this design - 2/3 of the resulting data stays more-or-less permanent over the run-time. Generating the name array over and over looks wasteful...

Reply:
Agreed, as we discussed - The function will return the pointer to the const list of class_id's to name and not assign the names each time.